components/TenantedLoginForm.tsx

```diff
@@ -1,17 +1,24 @@
-import { FormEventHandler, useEffect, useState } from "react";
+import { useEffect, useState } from "react";
 import { login } from "@/lib/api";
 import {
-  formatSSOStartURL,
-  Organization,
-  publicToken,
+    formatOAuthStartURL,
+    formatSSOStartURL,
+    Organization,
 } from "@/lib/loadStytch";
 import { EmailLoginForm } from "./EmailLoginForm";
+import Link from "next/link";
 
 type Props = {
   org: Organization;
 };
 const TenantedLoginForm = ({ org }: Props) => {
-  return (
+    const [googleOAuthURL, setGoogleOAuthURL] = useState("");
+    useEffect(() => {
+        // UseEffect since format requires window to be loaded
```

Comment:
You'll notice that SSO doesn't need to do this - SSO depends on using the `default` redirect URL for the project. That might be brittle & it might make sense to do this instead. 

If we want to go this route, can we use [`getInitialProps`](https://nextjs.org/docs/pages/api-reference/functions/get-initial-props) instead of the clientside rerender?

Something like
```
Page.getInitialProps = async (ctx: NextPageContext) => {
  const domain = getDomainFromWindow() || ctx.req?.host;
  return { domain };
};
```

If you'd like to do this - can you update SSO too?

Reply:
Agh feel like w/ the amount I'm referencing this it may be better to just do the work in the `OAuthButton` component. LMK if you think this is fine, but I kind of prefer taking away all the server side calls for domain and doing something like 


``` 
export const OAuthButton = ({ providerType, orgSlug }: Props) => {
    const [domain, setDomain] = useState<string | null>(null);
    useEffect(() => {
        setHostDomain(getDomainFromWindow());
    }, [])
    const isDiscovery = orgSlug == null;
    const oAuthStartURL = isDiscovery ? formatOAuthDiscoveryStartURL(domain, providerType) : formatOAuthStartURL(domain, providerType, orgSlug);
```